### PR TITLE
Persist corrupt docs

### PR DESCRIPTION
**Pull request by kekachmar**
In scribe, when we encounter a protocol error, it causes document corruption. However, when scribe restarts, we try to reprocess corrupted documents since the document corruption is stored in memory. This change is to persist the corrupted documents in our checkpoints (which are stored in the database) so that we can avoid this reprocessing.

### CODE REVIEW

File: server/routerlicious/packages/lambdas/src/scribe/lambda.ts [hunk 1]
```diff
@@ -445,20 +448,32 @@ export class ScribeLambda implements IPartitionLambda {
...
+	) {
 		// Get checkpoint context
 		const checkpoint = this.generateScribeCheckpoint(message.offset);
 		this.updateCheckpointMessages(message);
 
 		// write the checkpoint with the current up-to-date state
-		this.checkpoint(checkpointReason);
+		if (!bypassKafka) {
+			// kafka checkpoint
+			this.checkpoint(checkpointReason);
```
Comment (pradeepvairamani) [inline server/routerlicious/packages/lambdas/src/scribe/lambda.ts:474]: I am not sure if these comments are correct. this.checkpoint() only updates the checkpoint info right? It does not do the checkpointing itself. That happens in checkpointCore. And checkpoint core is where both the db checkpointing and the kafka checkpointing happens

File: server/routerlicious/packages/lambdas/src/scribe/lambda.ts [hunk 2]
```diff
@@ -518,6 +533,9 @@ export class ScribeLambda implements IPartitionLambda {
 			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
 			const message = this.pendingMessages.shift()!;
 			try {
+				if (!this.isDocumentCorrupt) {
+					throw new Error(`Test Error`);
```
Comment (pradeepvairamani) [inline server/routerlicious/packages/lambdas/src/scribe/lambda.ts:537]: Do we want to remove this?

Comment (kekachmar) [reply server/routerlicious/packages/lambdas/src/scribe/lambda.ts:537]: @pradeepvairamani - I am still working on this, meant to mark it as a draft.

File: server/routerlicious/packages/lambdas/src/scribe/lambda.ts (hunk 1 above)
Comment (kekachmar) [reply server/routerlicious/packages/lambdas/src/scribe/lambda.ts:474]: Yes, you are correct. I added this in the wrong spot. I meant to add this on `this.context.checkpoint`. Updating now

**Pull request closed without merging**